src/state/groups/actions.js

```diff
@@ -194,56 +152,57 @@ export const sendDeleteGroup = groupCode => dispatch => (
   })
 );
 
-export const fetchCurrentPageGroupDetail = () => (dispatch, getState) => (
-  new Promise((resolve) => {
-    const { groupname } = getParams(getState());
-    getGroup(groupname).then((response) => {
-      response.json().then((json) => {
-        if (response.ok) {
-          dispatch(setSelectedGroup(json.payload));
-          resolve();
-        } else {
-          dispatch(addErrors(json.errors.map(err => err.message)));
-          resolve();
-        }
-      });
-    });
-  })
-);
+export const setReferences = references => ({
+  type: SET_REFERENCES,
+  payload: {
+    references,
+  },
+});
 
-const fetchCurrentReference = (getApiCall, setActionCreator) =>
-  (page = { page: 1, pageSize: 10 }) => (dispatch, getState) => (
+export const fetchReferences = (referenceKey, page = { page: 1, pageSize: 10 }) =>
+  (dispatch, getState) => (
     new Promise((resolve) => {
       const { groupname } = getParams(getState());
       dispatch(toggleLoading('references'));
-      getApiCall(page, groupname).then((response) => {
+      getReferences(page, groupname, referenceKey).then((response) => {
         response.json().then((json) => {
           if (response.ok) {
-            dispatch(setActionCreator(json.payload));
-            dispatch(toggleLoading('references'));
+            dispatch(setReferences({
+              [referenceKey]: json.payload,
+            }));
             dispatch(setPage(json.metaData));
-            resolve();
           } else {
             dispatch(addErrors(json.errors.map(err => err.message)));
-            dispatch(toggleLoading('references'));
-            resolve();
           }
+          dispatch(toggleLoading('references'));
+          resolve();
         });
       });
     })
   );
 
-export const fetchCurrentReferencePages =
-  fetchCurrentReference(getPageReferences, setSelectedGroupPageReferences);
-
-export const fetchCurrentReferenceUsers =
-    fetchCurrentReference(getUserReferences, setSelectedGroupUserReferences);
-
-export const fetchCurrentReferenceWidgetTypes =
-        fetchCurrentReference(getWidgetTypeReferences, setSelectedGroupWidgetTypeReferences);
-
-export const fetchCurrentReferenceContents =
-        fetchCurrentReference(getContentReferences, setSelectedGroupContentReferences);
-
-export const fetchCurrentReferenceResources =
-        fetchCurrentReference(getResourceReferences, setSelectedGroupResourceReferences);
+export const fetchCurrentPageGroupDetail = () => (dispatch, getState) => (
+  new Promise((resolve) => {
+    const { groupname } = getParams(getState());
+    getGroup(groupname).then((response) => {
+      response.json().then((json) => {
```

Comment:
You might have a non-json response, please make sure it is before calling response.json(), or use a try/catch block to handle these kind of errors